nshutijonathan/Auto-Mart
# Challenge 2 feedback before presentation

#### What does this PR do?
- Implemented optional endpoints in auto-mart restful API
#### Description of Task to be completed?
- Have The following endpoints tested and working 
*   GET/ localhost:3000/api/v1/availablemanufacturer/cars?status=available&manufacturer=toyota
*   GET/localhost:3000/api/v1/type/cars?body_type=req.query
*   GET/localhost:3000/api/v1/range/cars?status=available&min_price=200&max_price=1000

#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* You must have node environment configured  and .env file in the root folder
* to test authentication and authorization, add this to the header 
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with s data structures  to persist some data
